via-router/src/cache.rs

```diff
@@ -1,29 +1,24 @@
 use std::collections::VecDeque;
+use std::error::Error;
+use std::fmt::{self, Display, Formatter};
 use std::sync::RwLock;
 
-use crate::router::Match;
+use crate::router::{Match, Matched};
+
+#[derive(Debug)]
+pub struct CacheError;
 
 pub struct Cache {
     capacity: usize,
-
-    #[allow(clippy::type_complexity)]
-    entries: RwLock<VecDeque<(Box<str>, Vec<Option<Match>>)>>,
+    entries: RwLock<VecDeque<(Box<str>, Matched)>>,
 }
 
-#[inline]
-fn fetch(
```

Comment:
To my understanding, there is nothing wrong with calling to_vec on a `&Vec<_>` in find_map since it is only called once if a match is found. That was the only reason for having a separate function here.